### ISSUE TITLE
KModifier.EXPECT type modifier is not propagated more than one TypeSpec deep

### ISSUE THREAD

**Issue by JustinMullin** [square/kotlinpoet#878]
I have a `FunSpec` nested underneath a few layers of `TypeSpec`s, with the top level being marked with the EXPECT modifier. When the `FunSpec` is more than two layers deep from the top-level type marked EXPECT, the generated method is being emitted with an empty body, rather than the expected declaration only.

A minimal example is the following:
```
TypeSpec.classBuilder("Outermost").addModifiers(KModifier.EXPECT)
    .addType(TypeSpec.classBuilder("Innermore")
        .addType(TypeSpec.classBuilder("Innermost")
            .addFunction(FunSpec.builder("foo").returns(ClassName("", "String")).build())
            .build())
        .build())
    .build()
```

Which generates:
```
expect class Outermost {
  class Innermore {
    class Innermost {
      fun foo(): String { // Compiler error due to implementation inside an `expect`
      }
    }
  }
}
```

The same example with one layer of TypeSpec removed:
```
TypeSpec.classBuilder("Outermost").addModifiers(KModifier.EXPECT)
    .addType(TypeSpec.classBuilder("Innermost")
        .addFunction(FunSpec.builder("foo").returns(ClassName("", "String")).build())
        .build())
    .build()
```

This does properly respect the outer EXPECT modifier, resulting in:
```
expect class Outermost {
  class Innermost {
    fun foo(): String
  }
}
```

**Comment by JustinMullin** [comment]
It appears https://github.com/square/kotlinpoet/issues/732 addressed this case previously for single-nesting of types, but does not address further nesting.

**Comment by ZacSweers** [comment]
Want to send a PR?

**Comment by Egorand** [comment]
Closed with #879
